code/modules/robotics/bot/secbot.dm

```diff
@@ -1156,7 +1088,77 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 
 		return
 
+//secbot handcuff bar thing
+/datum/action/bar/icon/secbot_cuff
+	duration = 60
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_STUNNED | INTERRUPT_ACTION
+	id = "secbot_cuff"
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "handcuff"
+	var/obj/machinery/bot/secbot/master
+
+	New(var/the_bot)
+		src.master = the_bot
+		..()
 
+	onUpdate()
+		..()
+		if (get_dist(master, master.target) <= 1 || !master.target || master.target.hasStatus("handcuffed"))
+			interrupt(INTERRUPT_ALWAYS)
+			master.process()
+			return
+
+	onStart()
+		..()
+		if (get_dist(master, master.target) <= 1 || !master.target || master.target.hasStatus("handcuffed"))
+			interrupt(INTERRUPT_ALWAYS)
+			master.process()
+			return
+
+		playsound(master, "sound/weapons/handcuffs.ogg", 30, 1, -2)
+		master.mode = SECBOT_ARREST
+		master.visible_message("<span class='alert'><B>[master] is trying to put handcuffs on [master.target]!</B></span>")
+
+	onEnd()
+		..()
+
+		if (get_dist(master, master.target) <= 1)
+			if (!master.target || master.target.hasStatus("handcuffed"))
+				return
+
+			var/uncuffable = 0
+			if (ishuman(master.target))
+				var/mob/living/carbon/human/H = master.target
+				if(!H.limbs.l_arm || !H.limbs.r_arm)
+					uncuffable = 1
+
+			if (!isturf(master.target.loc))
+				uncuffable = 1
+
+			if(ishuman(master.target) && !uncuffable)
+				master.target.handcuffs = new /obj/item/handcuffs(master.target)
+				master.target.setStatus("handcuffed", duration = INFINITE_STATUS)
+
+			var/last_target = master.target
+
+			master.mode = SECBOT_IDLE
+			master.target = null
+			master.anchored = 0
+			master.last_found = world.time
+			master.frustration = 0
+
+			if(!uncuffable)
+				playsound(master.loc, pick('sound/voice/bgod.ogg', 'sound/voice/biamthelaw.ogg', 'sound/voice/bsecureday.ogg', 'sound/voice/bradio.ogg', 'sound/voice/binsult.ogg', 'sound/voice/bcreep.ogg'), 50, 0, 0, 1)
+			if (master.report_arrests && !uncuffable)
+				var/bot_location = get_area(master)
+					//////PDA NOTIFY/////
+				var/datum/radio_frequency/transmit_connection = radio_controller.return_frequency("1149")
```

Comment:
I don't much like the frequency be hardcoded like this (same for the "officer down" message in Beepsky even though it's not a part of this PR). Consider moving it to /secbot just like the other frequencies are or maybe creating a global define for the PDA message frequency.

Reply:
I made a new define file for signal defs. Right now it just has the one, but its there and seems to work!